tensorlayer/layers/convolution.py

```diff
@@ -1153,6 +1154,82 @@ def __init__(
+
+    Parameters
+    ----------
+    prev_layer : :class:`Layer`
+        Previous layer with a 4D output tensor in the shape of (batch, height, width, channels).
+    shape : tuple of int
+        The shape of the filters: (filter_height, filter_width, out_channels, in_channels).
+    output_shape : tuple of int
+        Output shape of the deconvolution.
+    rate : int
+        The stride that we sample input values in the height and width dimensions.
+        This equals the rate that we up-sample the filters by inserting zeros across the height and width dimensions.
+        In the literature, this parameter is sometimes mentioned as input stride or dilation.
+    act : activation function
+        The activation function of this layer.
+    padding : str
+        The padding algorithm type: "SAME" or "VALID".
+    W_init : initializer
+        The initializer for the weight matrix.
+    b_init : initializer or None
+        The initializer for the bias vector. If None, skip biases.
+    W_init_args : dictionary
+        The arguments for the weight matrix initializer.
+    b_init_args : dictionary
+        The arguments for the bias vector initializer.
+    name : str
+        A unique layer name.
+
+    """
+
+    @deprecated_alias(layer='prev_layer', end_support_version=1.9)  # TODO remove this line for the 1.9 release
+    def __init__(
+            self, prev_layer, shape=(3, 3, 128, 256), output_shape=(1, 64, 64, 128), rate=2, act=tf.identity,
+            padding='SAME', W_init=tf.truncated_normal_initializer(stddev=0.02),
+            b_init=tf.constant_initializer(value=0.0), W_init_args=None, b_init_args=None, name='atrou2dtrans'
+    ):
+
+        super(AtrousConv2dTransLayer, self).__init__(prev_layer=prev_layer, name=name)
+        logging.info(
+            "AtrousConv2dTransLayer %s: shape:%s output_shape:%s rate:%d pad:%s act:%s" %
+            (name, shape, output_shape, rate, padding, act.__name__)
+        )
+
+        self.inputs = prev_layer.outputs
+
+        if W_init_args is None:
+            W_init_args = {}
+        if b_init_args is None:
+            b_init_args = {}
+        if act is None:
+            act = tf.identity
+
+        with tf.variable_scope(name):
+            filters = tf.get_variable(
+                name='filter', shape=shape, initializer=W_init, dtype=LayersConfig.tf_dtype, **W_init_args
+            )
+            if b_init:
+                b = tf.get_variable(
+                    name='b', shape=(shape[-2]), initializer=b_init, dtype=LayersConfig.tf_dtype, **b_init_args
+                )
+                self.outputs = act(tf.nn.atrous_conv2d_transpose(self.inputs, filters, output_shape, rate, padding) + b)
+            else:
+                self.outputs = act(tf.nn.atrous_conv2d_transpose(self.inputs, filters, output_shape, rate, padding))
```

Comment:
please use named parameters, this insure stability if the TF API changes, example: `my_function(a=1, b=2)`

Reply:
OK!